locator/topology.cc

```diff
@@ -573,16 +573,22 @@ void topology::sort_by_proximity(locator::host_id address, host_id_vector_replic
 }
 
 void topology::do_sort_by_proximity(locator::host_id address, host_id_vector_replica_set& addresses) const {
-    std::sort(addresses.begin(), addresses.end(), [this, &address](const locator::host_id& a1, const locator::host_id& a2) {
-        return compare_endpoints(address, a1, a2) < 0;
-    });
-}
-
-std::weak_ordering topology::compare_endpoints(const locator::host_id& address, const locator::host_id& a1, const locator::host_id& a2) const {
+    struct info {
+        locator::host_id id;
+        int distance;
+    };
+    utils::small_vector<info, host_id_vector_replica_set::internal_capacity()> host_infos;
+    host_infos.reserve(addresses.size());
     const auto& loc = get_location(address);
-    const auto& loc1 = get_location(a1);
-    const auto& loc2 = get_location(a2);
+    std::ranges::copy(addresses | std::views::transform([&] (locator::host_id id) {
+        const auto& loc1 = get_location(id);
+        return info{ id, distance(address, loc, id, loc1) };
+    }), std::back_inserter(host_infos));
```

Comment:
std::back_inserter() is slow if the compiler can't figure out the reservation is sufficient. std::ranges::to is likely to be better, since it uses the from_ranges_t protocol to initialize the elements in-place.

Reply:
Used std::ranges::to in v6 ([d1490bb](https://github.com/scylladb/scylladb/commit/d1490bb7bfcdabe36ff50d88379e7edb7cbe2ddf))